test/cliffordsynthesis/test_synthesis.cpp

```diff
@@ -319,4 +349,66 @@ TEST_P(SynthesisTest, TestDestabilizerTwoQubitGates) {
   }
 }
 
+TEST(HeuristicTest, basic) {
+  auto config = Configuration();
+  auto qc     = qc::QuantumComputation(2);
+  qc.h(0);
+  qc.s(1);
+  qc.h(0);
+  qc.s(1);
+  config.heuristic = true;
+  config.splitSize = 1;
+  config.target    = TargetMetric::Depth;
+  auto synth       = CliffordSynthesizer(qc);
+  synth.synthesize(config);
+  EXPECT_EQ(synth.getResults().getDepth(), 2);
+}
+
+TEST(HeuristicTest, identity) {
+  auto config = Configuration();
+  auto qc     = qc::QuantumComputation(2);
+  qc.h(0);
+  qc.s(1);
+  qc.h(0);
+  qc.sdag(1);
+  config.heuristic = true;
+  config.splitSize = 2;
+  config.target    = TargetMetric::Depth;
+  auto synth       = CliffordSynthesizer(qc);
+  synth.synthesize(config);
+  EXPECT_EQ(synth.getResults().getDepth(), 0);
+}
+
+using namespace qc::literals;
+TEST(HeuristicTest, threeLayers) {
+  auto config = Configuration();
+  auto qc     = qc::QuantumComputation(2);
+  qc.h(0);
+  qc.h(1);
+  qc.x(1, 0_pc);
+  qc.h(0);
+  qc.h(1);
+  config.heuristic = true;
+  config.splitSize = 2;
+  config.target    = TargetMetric::Depth;
+  auto synth       = CliffordSynthesizer(qc);
+  synth.synthesize(config);
+  EXPECT_EQ(synth.getResults().getDepth(), 3);
+}
+
+using namespace qc::literals;
```

Comment:
Instead of repeating the using directive here twice, consider moving it to the very top (only once 🙃)

Reply:
C&P mistake. Fixed it.